### DISCUSSION TITLE
osu!lazer doesn't start with Nvidia PRIME Performance Mode enabled (Linux Mint)

### DISCUSSION THREAD

**Discussion by SomeFafr** [ppy/osu#28754] [Q&A] [answered]
So, in the Nvidia X Server Settings I have enabled PRIME Performance Mode due to some issues with running games with Proton, but now osu!lazer doesn't open. Judging from the runtime logs, the problem seems to be that it tries to use the iGPU instead of the dGPU, but it's unavailable because of performance mode. Running the .appImage through the terminal gives this: 
`X Error of failed request:  BadMatch (invalid parameter attributes)`
`  Major opcode of failed request:  149 ()`
`  Minor opcode of failed request:  4`
`  Serial number of failed request:  273`
`  Current serial number in output stream:  329`
What can I do to make osu!lazer use the dGPU or allow it to use the iGPU despite performance mode being enabled? Would messing with application profiles be able to let me do that?

[1720294103.runtime.log](https://github.com/user-attachments/files/16117114/1720294103.runtime.log)
[1720294103.input.log](https://github.com/user-attachments/files/16117115/1720294103.input.log)
[1720294103.auth.log](https://github.com/user-attachments/files/16117117/1720294103.auth.log)

**Comment by Feodor0090** [comment, marked as answer]
According to logs the game attempts to run on igpu using vulkan. I'd start by changing renderer back to OGL via game's settings `ini` (`framework.ini` file -> `Renderer = OpenGL` line). Then launch it with nvidia's glx (i.e. `prime-run ./osu.AppImage`). Most likely this will be enough to make it work.

**Reply by SomeFafr** [reply]
Oh, that actually did help. Honestly, I forgot I even set the renderer to Vulkan before. Thank you!
